dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/CompanyManager.kt

```diff
@@ -200,6 +209,7 @@ class CompanyManager(
      * @param companyId the ID of the company to be checked
      * @return a boolean signalling if the company is public or not
      */
+    @Transactional
```

Comment:
I think it's better to remove this to avoid unnecessary transactional overhead.

The @Transactional annotation is primarily used to handle transactional behavior when performing database read and write operations. It ensures that the operations are executed within a transactional context and provides benefits such as data integrity, consistency, and automatic rollback in case of exceptions.

However, the `isCompanyPublic()` function only retrieves data from the database without making any modifications, which is why it might be unnecessary to use the @Transactional annotation.

This goes for all ready-only operations in this whole file.   Let's remove all @Transactional annotations there.

FYI: In read-only scenarios, Spring allows you to specify the readOnly attribute of the @Transactional annotation to optimize performance. By setting `readOnly = true`, you indicate that the transaction is read-only and can be optimized accordingly.   But of course we can also remove it completely in my opinion, given that I don't oversee another factor here.

What do you think?

Reply:
I added it here because using the annotation on private methods like `getCompanyById` is ignored. But I wanted to maintain the effect originally intended. Do you know if `readOnly = true` is preferrable over leaving the annotation out entirely? Probably only if there are multiple database calls.

Reply:
Leider weiß ich da auch nicht mehr. Da die Funktion aber sowieso nur ein DB Read ausführt, könnte die DB das im Zweifelsfall auch selbst herausfinden (und ich glaube nicht dass man hier das [AT]Transactional braucht.)

Reply:
Was ist mit den ganzen adneren `@Transactional` annotations hier?
Außer `addCompany()` sind das hier doch alles reine read-operations.
Können wir das daher nicht überall rausschmeißen? @MarcTM01  @fsoffel

Reply:
Schau ich drauf, wenn ich Zeit hab. Ansonsten ist das doch auch ein Fall für den Refactoring Log.

Reply:
Created a refactoring log item